src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.RenameTrackingCodeAction.cs

```diff
@@ -83,17 +81,15 @@ private bool TryInitializeRenameTrackingCommitter(CancellationToken cancellation
                         }
 
                         var snapshotSpan = stateMachine.TrackingSession.TrackingSpan.GetSpan(stateMachine.Buffer.CurrentSnapshot);
-                        var newName = snapshotSpan.GetText();
-                        var displayText = string.Format(EditorFeaturesResources.Rename_0_to_1, stateMachine.TrackingSession.OriginalName, newName);
-                        _renameTrackingCommitter = new RenameTrackingCommitter(stateMachine, snapshotSpan, _refactorNotifyServices, _undoHistoryRegistry, displayText);
+                        _renameTrackingCommitter = new RenameTrackingCommitter(stateMachine, snapshotSpan, _undoHistoryRegistry);
                         return true;
                     }
                 }
 
                 return false;
             }
 
-            private sealed class RenameTrackingCommitterOperation : CodeActionOperation
+            private class RenameTrackingCommitterOperation : RenameTrackingOperation
```

Comment:
We can't just get away with a regular apply changes operation now? Is the only thing happening now with the Commit() method that's special is we're clearing state, which is also now happening by the changes in TryApplyChanges?

Reply:
For testing purposes it's easier to work with the before/after solution, which will have the rename annotations. If we use TryApplyChanges the annotations are removed. We could modify the original operation to return the changed solution for tests, but this was the only place where we are directly getting the code action instead AND testing rename. I felt this was simpler to maintain and enforces a more correct behavior of not trying to use a code operation to test before/after solution changes.